src/conn/conn_cache_pool.c

```diff
@@ -453,22 +469,54 @@ __cache_pool_assess(WT_SESSION_IMPL *session, uint64_t *phighest)
 			continue;
 		cache = entry->cache;
 		++entries;
-		new = cache->bytes_read;
-		/* Handle wrapping of eviction requests. */
-		if (new >= cache->cp_saved_read)
-			cache->cp_current_read = new - cache->cp_saved_read;
+
+		/*
+		 * Figure out a delta since the last time we did an assessment
+		 * for each metric we are tracking.  Watch out for wrapping
+		 * of values.
+		 */
+		tmp = cache->bytes_read;
+		if (tmp >= cache->cp_saved_read)
+			reads = tmp - cache->cp_saved_read;
 		else
-			cache->cp_current_read = new;
-		cache->cp_saved_read = new;
-		if (cache->cp_current_read > highest)
-			highest = cache->cp_current_read;
+			reads = (UINT64_MAX - cache->cp_saved_read) + tmp;
```

Comment:
I had that originally, but it was more confusing to document why it was OK for the value to be inaccurate than to make it accurate.